qdax/tasks/brax_envs.py

```diff
@@ -264,125 +122,52 @@ def reset_based_scoring_function_brax_envs(
     environment, use "play_reset_fn = env.reset".
 
     To define purely deterministic environments, as in "scoring_function", generate
-    a single init_state using "init_state = env.reset(random_key)", then use
-    "play_reset_fn = lambda random_key: init_state".
+    a single init_state using "init_state = env.reset(key)", then use
+    "play_reset_fn = lambda key: init_state".
 
     Args:
         policies_params: The parameters of closed-loop controllers/policies to evaluate.
-        random_key: A jax random key
+        key: A jax random key
         episode_length: The maximal rollout length.
         play_reset_fn: The function to reset the environment and obtain initial states.
         play_step_fn: The function to play a step of the environment.
-        behavior_descriptor_extractor: The function to extract the behavior descriptor.
+        descriptor_extractor: The function to extract the descriptor.
 
     Returns:
         fitness: Array of fitnesses of all evaluated policies
         descriptor: Behavioural descriptors of all evaluated policies
         extra_scores: Additional information resulting from the evaluation
-        random_key: The updated random key.
     """
 
-    random_key, subkey = jax.random.split(random_key)
-    keys = jax.random.split(
-        subkey, jax.tree_util.tree_leaves(policies_params)[0].shape[0]
-    )
-    reset_fn = jax.vmap(play_reset_fn)
-    init_states = reset_fn(keys)
+    # Reset environments
+    key, subkey = jax.random.split(key)
+    keys = jax.random.split(subkey, jax.tree.leaves(policies_params)[0].shape[0])
+    init_states = jax.vmap(play_reset_fn)(keys)
 
-    fitnesses, descriptors, extra_scores, random_key = scoring_function_brax_envs(
-        policies_params=policies_params,
-        random_key=random_key,
-        init_states=init_states,
+    # Step environments
+    unroll_fn = partial(
+        generate_unroll,
         episode_length=episode_length,
         play_step_fn=play_step_fn,
-        behavior_descriptor_extractor=behavior_descriptor_extractor,
+        key=key,
```

Comment:
I do not think the `key` should be in the partial. At the moment, it seems the same key is used for all episodes.